### ISSUE TITLE
research how to filter bad 404 sites that are dead 

### ISSUE THREAD

**Issue by gbinal** [GSA/site-scanning#354]
* These are websites that return `target_url_404_test` = False, `final_url_live` = True and whose final URL's suggest that the site is dead (even though they return 2xx server codes).  

The best place to start with this is to look at all of the current websites that meet the first two of those criteria.

**Comment by gbinal** [comment]
Some notes as we research this.  Looking at the [2-21-23 snapshot](https://docs.google.com/spreadsheets/d/1teEahZtqt07-P4pXQqcdzdq8y_BXPgfhEJsy27EdUrs/edit#gid=1551781898), there's over 5k entries that  return `target_url_404_test` = False and `final_url_live` = True.  

In some cases, this is happening not because of ill-configured 404 but instead because the entire website or domain redirects to a live page.  E.g. `www.tsc.gov` redirects to `https://www.fbi.gov/investigate/terrorism/tsc`, but so does `www.tsc.gov/sklhslkhdslkshd`.

**Comment by gbinal** [comment]
Alright - so as I look through the data, it seems that the above issue is wayyyyyyyyyyyy more prominent than the problem I'm trying to solve for.  In fact, I struggled to find examples of the problem, even though I know that they exist.  

I finally found 3 problem sites when I filtered the Final URL field for `404` and found these: 
* save.uscis.gov
* hyperfueling.ksc.nasa.gov
* stm.foh.psc.gov

My only two ideas of how to better detect when this is happening are: 
* Search for whether `404` is in the final URL - though we've just seen that that's of very limited use.   
* Search the page source code for snippets such as `404`, `Page Not Found`, or `Cannot Be Found` and create a new data field that tracks that and allow one to filter for it.  

Having discussed this more with the team though, we're going to hold off on doing more work on this until a user actually asks about or complains about it.  I sometimes get ahead of myself and need to remember that we should focus on the feedback and requests we actually _are_ getting from our users and prioritize that work!  :)
